.github/workflows/linux-asan-tests.yml

```diff
@@ -0,0 +1,44 @@
+name: Linux ASAN Tests
+on:
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - '*.md'
+      - 'docs/**'
+  pull_request:
+    types: [opened, synchronize]
+    paths-ignore:
+      - '*.md'
+      - 'docs/**'
+  release:
+    types: [published]
+    paths-ignore:
+      - '*.md'
+      - 'docs/**'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
```

Comment:
It's good practice to specify the version by hash in case deps are compromised.

https://blog.rafaelgss.dev/why-you-should-pin-actions-by-commit-hash

Reply:
Done